### PR TITLE
add systemd-nspawn --volatile=overlay support, as well as the same for host systems 

### PR DESCRIPTION
**Pull request by poettering**
Let's add some magic for running nspawn containers with overlayfs as root. And similar for host systems.

### CODE REVIEW

File: man/systemd-nspawn.xml [hunk 1]
```diff
@@ -183,9 +183,16 @@
         <listitem><para>If specified, the container is run with a temporary snapshot of its file system that is removed
         immediately when the container terminates. May not be specified together with
         <option>--template=</option>.</para>
-        <para>Note that this switch leaves host name, machine ID and
-        all other settings that could identify the instance
-        unmodified.</para></listitem>
+        <para>Note that this switch leaves host name, machine ID and all other settings that could identify the
+        instance unmodified. Please note that — as with <option>--template=</option> — taking the temporary snapshot is
+        more efficient on file systems that support subvolume snapshots or 'reflinks' naively (<literal>btrfs</literal>
```
Comment (gdamjan) [inline man/systemd-nspawn.xml:188]:
```suggestion
        more efficient on file systems that support subvolume snapshots or 'reflinks' natively (<literal>btrfs</literal>
```

File: man/systemd-nspawn.xml [hunk 2]
```diff
@@ -183,9 +183,16 @@
...
+        or new <literal>xfs</literal>) than on more traditional file systems that do not
+        (<literal>ext4</literal>).</para>
+
+        <para>With this option no modifications of the container image run is retained after it terminated. Use
```
Comment (keszybz) [inline man/systemd-nspawn.xml]: Not sure what you mean here, maybe "no modifications of the container image are retained"?

File: man/systemd-nspawn.xml [hunk 3]
```diff
@@ -1074,33 +1085,49 @@
...
-        <para>Note that enabling this setting will only work correctly with operating systems in the container that can
-        boot up with only <filename>/usr</filename> mounted, and are able to automatically populate
-        <filename>/var</filename>, and also <filename>/etc</filename> in case of
-        <literal>--volatile=yes</literal>.</para></listitem>
+        <para>Note that setting this option to <option>yes</option> or <option>state</option> will only work correctly
+        with operating systems in the container that can boot up with only <filename>/usr</filename> mounted, and are
+        able to automatically populate <filename>/var</filename>, and also <filename>/etc</filename> in case of
+        <literal>--volatile=yes</literal>. The <option>overlay</option> option does not require any particular
+        preparations in the OS, but do note that <literal>overlayfs</literal> behaviour differs from regular file
+        systems in a number of ways, and hence compatibility might be limited.</para></listitem>
```
Comment (keszybz) [inline man/systemd-nspawn.xml]: "might be limited" → "is limited" (we know that it is, you just said so earlier in the sentence).

**Comment by poettering** [comment]
> ```shell
> sudo SYSTEMD_LOG_LEVEL=debug build/systemd-nspawn -xD / --volatile=overlay
> ```

Hmm, that's an issue with --ephemeral, and should be fixed, but independently, as it doesn't really have much to do with --volatile=

**Comment by poettering** [comment]
Force pushed a new version: only changes are the two man fixes. Setting green label, since changes are so minimal.

Will prep independent patch for the -x thing.

**Comment by poettering** [comment]
Prepped a separate patch for the -x thing:  #11863 (and don't be confused, there are only two relevenat commits in that PR, everything else is just this PR #11243, which I based the PR on since it rewords some paragraph also changed in this PR)

File: man/systemd-nspawn.xml (hunk 1 above)
Comment (poettering) [reply man/systemd-nspawn.xml:188]: Forgot to fix this before the merge, but #11863 finally fixes this one too!

**Pull request merged**